spec/app/ics-020-fungible-token-transfer/README.md

```diff
@@ -44,6 +44,7 @@ interface FungibleTokenPacketData {
   sender: string
   receiver: string
   memo: string
+  forwardingPath: string
```

Comment:
I keep thinking that forwarding should be combined with execution (i.e.: ibc-hooks/adr-8) so we can define a more generic declarative model of computation on top of ibc. The simplest solution is to allow a callback to exist on each hop as long as it returns something that can be processed on the next hop (new packet, short-circuit, or maybe even a modified path). This would be very similar to a functional `fold` or, more generally, a combination of monads. 

A more advanced version of this may a allow for fan-in/out, but it may be better to leave that for the future as I think the fold model covers 90% of the use cases. 

Does it make sense to try to integrate those directly on this spec? I've been meaning to write a document explaining this in more detail but haven't gotten around to it.